Automattic/jetpack
# Protect: Add "new" badge for users who have never viewed the Firewall tab

This PR adds a "new" badge to the Firewall tab label, when the current user has never opened that tab.

#### Changes proposed in this Pull Request:

* Adds two API endpoints to fetch and update a `jetpack_protect_waf_seen` user meta value.
* Integrates the above into a `wafSeen` redux store value.
* Adds the "new" badge to the Firewall tab.

#### Other information:


#### Jetpack product discussion



1202232924268038-as-1203279663034176

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Spin up a new Jurassic Ninja site running this branch via Jetpack Beta:
https://jurassic.ninja/create?jetpack-beta&branches.jetpack-protect=add/protect-waf-tab-new-badge
* View the Protect admin dashboard and verify the "new" badge is displayed.
* Open the Firewall tab and verify the badge is removed.
* Refresh the page and verify the badge is no longer shown.
* Monitor the network tab during this process to verify:
  * The `waf-seen` GET request is only sent once per session.
  * The `waf-seen` POST request is only sent once per session, when the Firewall tab is displayed.

#### Screenshots

![image](https://user-images.githubusercontent.com/10933065/199853505-0ddf4c1c-8c48-4f2b-b7a0-b603169b8133.png)
